### ISSUE TITLE
GAMA is blocked when opening a display in full screen

### ISSUE THREAD

**Issue by benoitgaudou** [gama-platform/gama#2386]
### Steps to reproduce
1. Run a model with a display (tested with a Java2D display).

2. click on the full screen button

### Actual behavior
GAMA totally freezes. The user cannot move back to not full screen.
Some people have also experienced a block of the mouse...

### System and version 
GAMA 1.8 RC, Ubuntu 17  or 16

**Comment by hqnghi88** [comment]
Waiting for the fix, does the ctrl+shift+X help? or it blocks also the keys?

**Comment by benoitgaudou** [comment]
Hi,
the keys were also blocked.
So nothing was possible ...

Benoit

**Comment by benoitgaudou** [comment]
Hi,
Here are some logs related to this bug, don't know whether it can help.

[hs_err_pid3228.log](https://github.com/gama-platform/gama/files/1893227/hs_err_pid3228.log)
[hs_err_pid17009.log](https://github.com/gama-platform/gama/files/1893228/hs_err_pid17009.log)

**Comment by AlexisDrogoul** [comment]
The logs say `OpenJDK Runtime Environment (8.0_151-b12)`. AFAIK, GAMA is not certified to work with OpenJDK. Can you try with Oracle's one ?

**Comment by AlexisDrogoul** [comment]
Have you had the time to test it with the official Java VM ?

**Comment by agrignard** [comment]
Just in case I tried with a recent version (11/08/18) #2290 is still happenning

**Comment by AlexisDrogoul** [comment]
@agrignard on Ubuntu ? And with which JDK ?

**Comment by agrignard** [comment]
me on Ubuntu ;-) ? 

No no it's happening on my mac (mac 10.11.6) and java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)

**Comment by agrignard** [comment]
Ok I thought #2290 was related to this one but maybe not.

**Comment by AlexisDrogoul** [comment]
@hqnghi88 can you try opening a display in fullscreen with the Oracle JVM ? I'm about to close the issue as the logs indicate OpenJDK, but wants to make sure that it works for the "official" JVM

**Comment by hqnghi88** [comment]
Dear Alexis,
It works like a charm, with shortcut key Esc and the button

**Comment by AlexisDrogoul** [comment]
Perfect. Case closed. Thank you @hqnghi88 !

**Comment by hqnghi88** [comment]
Well, java2d have problem. I missed the case

**Comment by AlexisDrogoul** [comment]
You mean you tested it with OpenGL displays ?

**Comment by hqnghi88** [comment]
Yes opengl work well, but the root post of benoit said Java2D. And i 've just restart pc

**Comment by hqnghi88** [comment]
Ok, At least, it does not block whole system, i can use system shortcut key like AltTab, StartTab to switch to eclipse.

**Comment by AlexisDrogoul** [comment]
I think the two issues (this one and #2379) are linked. Java2D displays seem to capture the mouse and keyboard events on GTK-based windowing systems (like Ubuntu), while they don't under MacOSX and Windows. Two strategies are then possible: find a way to "un-focus" them (i.e. to never let them have the keyboard and mouse focus) or define the corresponding AWT listeners (which would, then, be only installed on GTK environments). I don't know exactly what to do. Any thoughts ?
